### DISCUSSION TITLE
Clean up Feature Requests

### DISCUSSION THREAD

**Discussion by equt** [tusen-ai/naive-ui#3542] [General]
- 🤔 indicates the description is vague
- 😇 indicates the feature request is hard to implement
- ♿️ A11y
- 🤓 Upstream

### Anchor

- https://github.com/tusen-ai/naive-ui/issues/2151 Smooth

### Auto Complete

- https://github.com/tusen-ai/naive-ui/issues/1459 emit `search` like `<NSelect />`

### Card

- https://github.com/tusen-ai/naive-ui/issues/1550 Secondary header slot

### Cascader

- https://github.com/tusen-ai/naive-ui/issues/2351 🤔

### Color Picker

- https://github.com/tusen-ai/naive-ui/issues/2844 `clearable`

### Data Table

- https://github.com/tusen-ai/naive-ui/issues/555 Dedicated table animation time variable
- https://github.com/tusen-ai/naive-ui/issues/1357 `cell-props` similar to the currently existing `row-props`
- https://github.com/tusen-ai/naive-ui/issues/1340 Horizontal virtual scroll
- https://github.com/tusen-ai/naive-ui/issues/3165 Resizable columns
- https://github.com/tusen-ai/naive-ui/issues/1563 Row reordering (currently through DnD)
- https://github.com/tusen-ai/naive-ui/issues/2814 Sticky table summary row at the bottom
- https://github.com/tusen-ai/naive-ui/issues/1555 Sticky table header row at the top
- https://github.com/tusen-ai/naive-ui/issues/2490 Hide table header
- https://github.com/tusen-ai/naive-ui/issues/1573 Expose scrollbar configuration
- https://github.com/tusen-ai/naive-ui/issues/651 Cell border

### Date Picker

- https://github.com/tusen-ai/naive-ui/issues/1178 Select only one value in `daterange`
- https://github.com/tusen-ai/naive-ui/issues/2240 Hide input text field
- https://github.com/tusen-ai/naive-ui/issues/2591 Custom time-zone support
- https://github.com/tusen-ai/naive-ui/issues/3167 Auto close year-month popover

### Dialog

- https://github.com/tusen-ai/naive-ui/issues/2152 Draggable
- https://github.com/tusen-ai/naive-ui/issues/2062 Promise to close
- https://github.com/tusen-ai/naive-ui/issues/2041 Count reference to all popup, e.g., notifications, message
 Promise to close popup

### Drawer

- https://github.com/tusen-ai/naive-ui/issues/505 `loading` option like the data table has
- https://github.com/tusen-ai/naive-ui/issues/2846 Configurable mask

### Dropdown

- https://github.com/tusen-ai/naive-ui/issues/2478 Function to determine `disabled` state

### Dynamic Input

- https://github.com/tusen-ai/naive-ui/issues/2749 DnD to sort

### Ellipsis

- https://github.com/tusen-ai/naive-ui/issues/75 More customizations 🤔 😇

### Form

- https://github.com/tusen-ai/naive-ui/issues/1743 `readonly` on `<NForm />`
- https://github.com/tusen-ai/naive-ui/issues/2302 Custom validation styles
- https://github.com/tusen-ai/naive-ui/issues/2675 Reactive validation result

### Image

- https://github.com/tusen-ai/naive-ui/issues/283 Lazy loading 😇
- https://github.com/tusen-ai/naive-ui/issues/1463 Fullscreen
- https://github.com/tusen-ai/naive-ui/issues/2029 `<picture>` and `<source>` tag
- https://github.com/tusen-ai/naive-ui/issues/3231 `on-preview-open` & `on-preview-close` event

### Input

- https://github.com/tusen-ai/naive-ui/issues/1385 Custom mask

### Layout

- https://github.com/tusen-ai/naive-ui/issues/1262 Drawer like `<NLayoutContent />` for slots

### Loading Bar

- https://github.com/tusen-ai/naive-ui/issues/2103 More granular control

### Menu

- https://github.com/tusen-ai/naive-ui/issues/1221 Shrinkable menu
- https://github.com/tusen-ai/naive-ui/issues/3092 Allow parent menu to be selected

### Modal

- https://github.com/tusen-ai/naive-ui/issues/490 Modal appears & leaves transition time customization
- https://github.com/tusen-ai/naive-ui/issues/1414 Draggable
- https://github.com/tusen-ai/naive-ui/issues/2261 Configurable mask
- https://github.com/tusen-ai/naive-ui/issues/2807 Maximize & Minimize

### Pagination

- https://github.com/tusen-ai/naive-ui/issues/583 `<NPagination />` supports `tiny` size
- https://github.com/tusen-ai/naive-ui/issues/2443 Pagination background 🤔

### Provider

- https://github.com/tusen-ai/naive-ui/issues/356 Default global `size` property
- https://github.com/tusen-ai/naive-ui/issues/1627 Spin stroke 
- https://github.com/tusen-ai/naive-ui/issues/2848 Allow to set global color transition time

### Progress

- https://github.com/tusen-ai/naive-ui/issues/136 `<NProgressBarGroup />`
- https://github.com/tusen-ai/naive-ui/issues/508 Spin
- https://github.com/tusen-ai/naive-ui/issues/509 Possible duplication of #508 🤔

### Popover

- https://github.com/tusen-ai/naive-ui/issues/544 ESC should close the popover ♿️
- https://github.com/tusen-ai/naive-ui/issues/545 Autofocus on confirm ♿️

### Radio

- https://github.com/tusen-ai/naive-ui/issues/1126 Vertical radio group

### Select

- https://github.com/tusen-ai/naive-ui/issues/300 Better keyboard interaction 🤔
- https://github.com/tusen-ai/naive-ui/issues/500 `clearable` button clear intermediate input content (e..g., creating new tags)

### Tabs

- https://github.com/tusen-ai/naive-ui/issues/1466 Tabs placement
- https://github.com/tusen-ai/naive-ui/issues/2128 Overflow behavior

### Time Picker

- https://github.com/tusen-ai/naive-ui/issues/390 `daterange` like `timerange` for the time picker
- https://github.com/tusen-ai/naive-ui/issues/715 ISO 8601

### Tooltip

- https://github.com/tusen-ai/naive-ui/issues/1992 Light theme

### Tree & Tree Select

- https://github.com/tusen-ai/naive-ui/issues/2477 Expose `pendingNodeKeyRef`
- https://github.com/tusen-ai/naive-ui/issues/1432 Partial update 🤓
- https://github.com/tusen-ai/naive-ui/issues/765 🤔
- https://github.com/tusen-ai/naive-ui/issues/2820 Transfer-like DnD 😇

### Upload

- https://github.com/tusen-ai/naive-ui/issues/1768 Pure front-end data handling

---

### New Components / Hooks

- https://github.com/tusen-ai/naive-ui/issues/85 General toolbar
- https://github.com/tusen-ai/naive-ui/issues/1471 Floating menu
- https://github.com/tusen-ai/naive-ui/issues/1598 `useLoading`
- https://github.com/tusen-ai/naive-ui/issues/1617 Preview, extracted from `<NUpload />`
- https://github.com/tusen-ai/naive-ui/issues/1111 Draggable, resizable card
- https://github.com/tusen-ai/naive-ui/issues/2535 Drag to sort list
- https://github.com/tusen-ai/naive-ui/issues/2047 Extract panel from `<NCascader />`
- https://github.com/tusen-ai/naive-ui/issues/2895 `<NModalProvider />`

---

### MISC

- https://github.com/tusen-ai/naive-ui/issues/2900 Atomic CSS
- https://github.com/tusen-ai/naive-ui/issues/2901 Native events
- https://github.com/tusen-ai/naive-ui/issues/3225 3rd codec
- https://github.com/tusen-ai/naive-ui/issues/2436 Partial refresh of subtree
- https://github.com/tusen-ai/naive-ui/issues/2119 Specify the virtual list item's height
- https://github.com/tusen-ai/naive-ui/issues/2381 Text gets cut off in popover like `<NMenu />` or `<NCascader />`
- https://github.com/tusen-ai/naive-ui/issues/3001 `name` field on all form items

Implemented?

- https://github.com/tusen-ai/naive-ui/issues/358
- https://github.com/tusen-ai/naive-ui/issues/1681
- https://github.com/tusen-ai/naive-ui/issues/1849
- https://github.com/tusen-ai/naive-ui/issues/2927

**Comment by equt** [comment]
准备花一段时间把现在还开着的 issues 理一下

- 有重复的就关掉，对于那些覆盖面特别大的，比如 https://github.com/tusen-ai/naive-ui/issues/1603 尽可能拆到小的 issues 上
- ~~打上独立的 `n-xxx` 标签，考虑在新建的时候加上相关的选项~~

**Comment by equt** [comment]
_reserved_

**Comment by equt** [comment]
Keywords

- **Draggable** - draggable popovers, like dialog or modal; drag to sort lists, table columns or rows
- **Mask Customization** - Mask for modal, drawer
- **Centralized Control of Popover** - Ref to active (on-screen) modal, notifications

**Comment by kamatil-dev** [comment]
#4239 Open unpreviewable items (other than images) in new tab. for example: docx, pdf....

**Comment by jahnli** [comment]
ref  https://github.com/tusen-ai/naive-ui/issues/6734
